teamcity/tests/test_teamcity.py

```diff
@@ -1,126 +1,110 @@
 import pytest
-from mock import ANY, MagicMock, patch
+from mock import ANY, patch
+from six import PY2
 
-from datadog_checks.teamcity import TeamCityCheck
+from datadog_checks.teamcity.constants import (
+    SERVICE_CHECK_BUILD_PROBLEMS,
+    SERVICE_CHECK_BUILD_STATUS,
+    SERVICE_CHECK_TEST_RESULTS,
+)
+from datadog_checks.teamcity.teamcity_rest import TeamCityRest
+
+from .common import (
+    BUILD_STATS_METRICS,
+    BUILD_TAGS,
+    EXPECTED_SERVICE_CHECK_TEST_RESULTS,
+    LEGACY_REST_INSTANCE,
+    NEW_FAILED_BUILD,
+    NEW_SUCCESSFUL_BUILD,
+    USE_OPENMETRICS,
+    get_fixture_path,
+)
 
-from .common import CHECK_NAME, CONFIG
+pytestmark = [
+    pytest.mark.skipif(USE_OPENMETRICS, reason='Tests not available in OpenMetrics version of check'),
+    pytest.mark.integration,
+]
 
 
-@pytest.mark.integration
-def test_build_event(aggregator):
-    teamcity = TeamCityCheck(CHECK_NAME, {}, [CONFIG])
+def test_build_event(dd_run_check, aggregator, rest_instance):
+    rest_instance['build_config_metrics'] = False
+    rest_instance['tests_health_check'] = False
+    rest_instance['build_problem_health_check'] = False
 
-    with patch('datadog_checks.base.utils.http.requests.get', get_mock_first_build):
-        teamcity.check(CONFIG['instances'][0])
+    teamcity = TeamCityRest('teamcity', {}, [rest_instance])
+    event_tags = BUILD_TAGS + ['instance_name:SampleProject_Build'] if PY2 else BUILD_TAGS
 
-    assert len(aggregator.metric_names) == 0
-    assert len(aggregator.events) == 0
-    aggregator.reset()
+    responses = json.load(open(get_fixture_path('event_responses.json'), 'r'))
+
+    single_build_responses = [
+        responses['build_config'],
+        responses['build_config_settings'],
+        responses['last_build'],
+        responses['new_builds'],
+        responses['server_details'],
+    ]
 
-    with patch('datadog_checks.base.utils.http.requests.get', get_mock_one_more_build):
-        teamcity.check(CONFIG['instances'][0])
+    multi_build_responses = [
+        responses['projects'],
+        responses['build_configs'],
+        responses['build_config_settings'],
+        responses['last_build'],
+        responses['new_builds'],
+        responses['server_details'],
+    ]
 
-    events = aggregator.events
-    assert len(events) == 1
-    assert events[0]['msg_title'] == "Build for One test build successful"
-    assert (
-        events[0]['msg_text']
-        == "Build Number: 2\nDeployed To: buildhost42.dtdg.co\n\nMore Info: "
-        + "http://localhost:8111/viewLog.html?buildId=2&buildTypeId=TestProject_TestBuild"
+    json_responses = single_build_responses if PY2 else multi_build_responses
+
+    with mock.patch('datadog_checks.base.utils.http.requests') as req:
+        mock_resp = mock.MagicMock(status_code=200)
+        mock_resp.json.side_effect = json_responses
+        req.get.return_value = mock_resp
+        teamcity.check(rest_instance)
+
+    assert len(aggregator.metric_names) == 0
```

Comment:
```suggestion
    assert not aggregator.metric_names
```
More pythonic:
https://peps.python.org/pep-0008/#programming-recommendations (search for "for sequences")